uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-calendar.component.spec.js

```diff
@@ -510,39 +560,70 @@
           controller.refresh('contacts');
           $rootScope.$digest();
 
-          simulateMonthsWithSignal('destroyed', controller.months.length);
+          simulateMonthWithSignal('destroyed', controller.months.length);
         }
       });
 
-      describe('selected months watcher', function () {
-        describe('when some other months are selected', function () {
-          beforeEach(function () {
-            controller.selectedMonths = [1, 2, 3];
-            $rootScope.$digest();
-          });
+      describe('selectCurrentMonth()', function () {
+        var currentAbsencePeriod;
 
-          it('sends the "show months" event with the newly selected months', function () {
-            expect($rootScope.$emit).toHaveBeenCalledWith(
-              'LeaveCalendar::showMonths',
-              controller.months.filter(function (month) {
-                return _.includes([1, 2, 3], month.index);
-              }),
-              jasmine.any(Boolean)
-            );
-          });
+        beforeEach(function () {
+          currentAbsencePeriod = _.find(controller.absencePeriods,
+            { current: true });
+          controller.injectMonth = false;
+          controller.selectedPeriod = controller.absencePeriods[1];
+          $rootScope.$digest();
+          controller.selectedMonthIndex = currentYear + '-' + (currentMonth + 1);
+          $rootScope.$digest();
+          controller.selectCurrentMonth();
+          $rootScope.$digest();
         });
 
-        describe('when none of the months are selected', function () {
-          beforeEach(function () {
-            controller.selectedMonths = [];
-            $rootScope.$digest();
-          });
+        it('sets the selected month as the current month', function () {
+          expect(controller.selectedMonth.year).toBe(currentYear);
+          expect(controller.selectedMonth.month).toBe(currentMonth);
+        });
+
+        it('sets the current absence period', function () {
+          expect(controller.selectedPeriod).toEqual(currentAbsencePeriod);
+        });
+      });
+
+      describe('paginateMonth()', function () {
+        var currentlySelectedMonth;
+        var tests = [
+          { direction: 'previous', monthDifference: -1 },
+          { direction: 'next', monthDifference: 1 }
+        ];
+
+        beforeEach(function () {
+          currentlySelectedMonth = controller.selectedMonth.month;
+          // This is needed to test the paginators availability
+          controller.months = _.slice(controller.months, 0, 3);
+        });
+
+        tests.forEach(function (test) {
+          describe('when user paginates to the ' + test.direction + ' month', function () {
```

Comment:
what's the difference between this test and the one a couple of lines above about pagination?

Reply:
The lines above describe how paginators behave on the component initiation.